applications/serial_lte_modem/src/slm_util.c

```diff
@@ -173,3 +172,25 @@ bool util_get_ipv4_addr(char *address)
 	address[tmp2 - tmp1] = '\0';
 	return true;
 }
+
+/**
+ * @brief Get string value from AT command with length check
+ */
+int util_string_get(const struct at_param_list *list, size_t index,
+			 char *value, size_t *len)
+{
+	int ret;
+	size_t size = *len;
+
+	/* at_params_string_get calls "memcpy" instead of "strcpy" */
+	ret = at_params_string_get(list, index, value, len);
+	if (ret) {
+		return ret;
+	}
+	if (*len < size) {
+		*(value + *len) = '\0';
```

Comment:
Shouldn't `len` be incremented since NULL termination is added?

Reply:
We're simulating strcpy(), and "len" is not used outside this function.

Reply:
We're simulating strcpy() just to help strlen() after the function call. Also outside the function, the "len" is only directly used after parsing the data to send uplink, in which context we should not include the null-terminator and thus not incremented

Reply:
You're right, this is misleading. Updated